crates/neon/src/types_impl/bigint.rs

```diff
@@ -0,0 +1,377 @@
+//! Types for working with [`JsBigInt`].
+
+use std::{error, fmt, mem::MaybeUninit};
+
+use crate::prelude::NeonResult;
+use crate::{
+    context::Context,
+    handle::Handle,
+    result::ResultExt,
+    sys::{self},
+    types::JsBigInt,
+};
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+/// Indicates if a `JsBigInt` is positive or negative
+pub enum Sign {
+    Positive,
+    Negative,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+/// Indicates a lossless conversion from a [`JsBigInt`] to a Rust integer
+/// could not be performed.
+///
+/// Failures include:
+/// * Negative sign on an unsigned int
+/// * Overflow of an int
+/// * Underflow of a signed int
+pub struct Error<T>(T);
```

Comment:
Why don't we call this `RangeError` to be more precise. For better or worse (both, really) we've avoided the Rust convention of overloading the name `Error` in Neon.